web/profiles/joinup/joinup.behat.inc

```diff
@@ -0,0 +1,163 @@
+<?php
+
+/**
+ * @file
+ * Contains \UserSubContext.
+ */
+
+use Behat\Gherkin\Node\TableNode;
+use Drupal\DrupalExtension\Context\DrupalSubContextBase;
+use Drupal\DrupalExtension\Context\DrupalSubContextInterface;
+use \Drupal\user\Entity\User;
+use \Drupal\file\Entity\File;
+
+/**
+ * Behat step definitions for testing users.
+ */
+class UserSubContext extends DrupalSubContextBase implements DrupalSubContextInterface {
+
+  /**
+   * Test users.
+   *
+   * @var \Drupal\user\Entity\User[]
+   */
+  protected $users = [];
+
+  /**
+   * Test images.
+   *
+   * @var \Drupal\file\Entity\File[]
+   */
+  protected $images = [];
+
+  /**
+   * Creates a user with data provided in a table.
+   *
+   * The following fields are required:
+   *  - Username
+   *  - Password
+   *  - E-mail
+   *  - First name
+   *  - Last name.
+   *
+   * @param TableNode $user_table
+   *   The collection data.
+   *
+   * @throws \Exception
+   *   Thrown when a column name is incorrect.
+   *
+   * @Given (the following )user:
+   */
+  public function givenUser(TableNode $user_table) {
+    $values = [];
+    $aliases = self::userFieldAliases();
+    // Replace the column aliases with the actual field names.
+    foreach ($user_table->getRowsHash() as $key => $value) {
+      if (array_key_exists($key, $aliases)) {
+        $values[$aliases[$key]] = $value;
+      }
+      else {
+        throw new \Exception("Unknown column '$key' in user table.");
+      }
+    };
+    $this->createUser($values);
+  }
+
+  /**
+   * Mapping of human readable names to machine names.
+   *
+   * @return array
+   *    Field mapping.
+   */
+  private static function userFieldAliases() {
+    return [
+      'Username' => 'name',
+      'Password' => 'pass',
+      'E-mail' => 'mail',
+      'Status' => 'status',
+      'First name' => 'field_user_first_name',
+      'Family name' => 'field_user_family_name',
+      'Photo' => 'field_user_photo',
+      'Professional profile' => 'field_user_professional_profile',
+      'Organisation' => 'field_user_organisation',
+      'Nationality' => 'field_user_nationality',
+      'Professional domain' => 'field_user_professional_domain',
+      // @todo Social network
+    ];
+  }
+
+  /**
+   * Creates a user from the given property and field data.
+   *
+   * @param array $values
+   *   An optional associative array of values, keyed by property name.
+   *
+   * @return \Drupal\user\Entity\User
+   *   A new user entity.
+   *
+   * @throws \Exception
+   *   Thrown when a given image is not found.
+   */
+  public function createUser(array $values) {
```

Comment:
You are overriding the createUser function here. Is this because it does not handle the image upload correctly?